src/vs/workbench/contrib/bulkEdit/browser/bulkFileEdits.ts

```diff
@@ -172,7 +172,7 @@ class DeleteOperation implements IFileOperation {
 			}
 		}
 
-		const useTrash = this._fileService.hasCapability(this.oldUri, FileSystemProviderCapabilities.Trash) && this._configurationService.getValue<boolean>('files.enableTrash');
+		const useTrash = this._fileService.hasCapability(this.oldUri, FileSystemProviderCapabilities.Trash) && this._configurationService.getValue<boolean>('files.enableTrash') && !this.options.doNotUseTrash;
```

Comment:
Is there any advantage in checking the option first before running hasCapability? e.g. 

Switch the clauses from: 

`const useTrash = this._fileService.hasCapability(this.oldUri, FileSystemProviderCapabilities.Trash) && this._configurationService.getValue<boolean>('files.enableTrash') && !this.options.doNotUseTrash;`

to 

`const useTrash =  !this.options.doNotUseTrash && this._fileService.hasCapability(this.oldUri, FileSystemProviderCapabilities.Trash) && this._configurationService.getValue<boolean>('files.enableTrash');`

I believe switching the above will prevent the function from being called if the option is true.

Reply:
Makes somewhat sense. Thanks